microsoft/iot-cardboard-js
# Query builder UI + field validation

### Summary of changes 🔍 
Initial PR for the UI of the query builder in Advanced Search modal. This new component will allow users to build queries based on properties, operators and values. This can be achieved by adding/removing query rows and updating the inputs in the respective fields. The scope of this PR includes property combo-box, operators changing based on property type and basic value field validation and row to string query creation.

This does not accomplish the following tasks for advanced search:
- Value auto-complete
- Query search and results integration
- SQL display of query

### Testing 🧪
- Open Advanced search base story
- Click on Add/Remove query row buttons
  -   Validate limit of 10 rows
  -   Validate first row not removable
- Select properties in property dropdown to see how operators change
  -   Validate string/numeric/boolean types to have different operators
- Add values in the respective input field to see how validation works:
  -   Validate empty string
  -   Validate no string values in numeric types

### Checklist ✔️
